src/projectDB/connection.h

```diff
@@ -0,0 +1,124 @@
+     * method does not allow the user to have nested transactions, for which
+     * they will need to use appropriate checkpoints using SQLite commands.
+     * @return True if a transaction was started successfully.
+     */
+    bool begin();
+
+    /**
+     * @brief Commit pending changes to the database.
+     * @details No modification operations are truly reflected in a SQLite
+     * database until a commit operation is performed. This method should be
+     * called after making any changes to the database, especially if the
+     * subsequent set of operations rely on these changes. Any outstanding
+     * modification that are left uncommitted are committed when their
+     * respective statements are finalized, something that happens whenever
+     * the Connection to the database is closed. This ensures that all changes
+     * made through a Connection during its lifetime are actually written to
+     * the database.
+     * @return True if the commit operation was executed successfully.
+     */
+    bool commit();
+
+    /**
+     * @brief Rollback any pending changes.
+     * @details This method effectively undoes any changes that have not been
+     * committed and are no longer desired. It should be noted that, while
+     * the SQLite library by default rolls back all pending changes when a DB
+     * connection is closed, this wrapper class will indirectly commit pending
+     * changes before closing the connection. So if any undesired changes are
+     * to be rolled back, they have to be explicitly done so before the
+     * destruction of a Connection object. Commits are a default over rollback
+     * because more often changes need to be committed than rolled back. The
+     * latter is assumed to be an active decision.
+     * @return True if the rollback operation was executed successfully.
+     */
+    bool rollback();
+
+    /**
+     * @brief Prepare a SQL statement and return a Cursor ready to be
+     * executed. See documentation of Cursor class for details.
+     * @param query A SQL query as a standard string.
+     * @return Pointer to a Cursor object that has to be executed/iterated upon.
+     */
+    Cursor* prepare(const std::string& query);
+
+    /**
+     * @brief Obtain the ROWID for the last row inserted.
+     * @details A ROWID in SQLite is available for all rows in a regular tables
+     * (except the ones created using the "WITHOUT ROWID" clause). This value
+     * is unique for each row in the table and if the table has a primary key
+     * integer column then that column becomes an alias for ROWID.
+     * @return An integer having the ROWID of last inserted row.
+     */
+    int lastInsertId();
+
+    /**
+     * @brief Obtain the absolute path to the connected database.
+     * @return Path as a string.
+     */
+    std::string dbPath();
+
+private:
+    /**
+     * @brief Stores absolute path for the connected SQLite database file.
+     */
+    std::string _dbPath;
```

Comment:
Since this is a private member and you are providing a public function to allow others to access it, this should be declared as a `const`. We don't want it to be modified by others right ?

Reply:
There is no setter for this variable, so I don't think there is a way of modifying this from anywhere outside the class.

Reply:
> there is no setter for this variable, so I don't think there is a way of modifying this from anywhere outside the class. 

Technically yes, no other class will be able to modify the variable. However, what I meant was,we never actually want  any other class  to change the Database path . So, even the copy should be immutable.

Reply:
Unless you know  how to implement this correctly, lets leave this as it is for now. 
I don't understand how to implement it  correctly. Sorry for the trouble